ClearKeep/Scenes/Search/ViewModel/SearchMessageViewModel.swift

```diff
@@ -0,0 +1,136 @@
+	public var isQuoteMessage: Bool
+	public var isForwardedMessage: Bool
+	public var isImageMessage: Bool
+	public var isFileMessage: Bool
+	public var clientWorkspaceDomain: String
+	public var avatar : String
+	public var groupName: String
+
+	public init(data: RealmMessage, members: [IMemberModel], group: String) {
+		id = data.messageId
+		groupId = data.groupId
+		groupType = data.groupType
+		fromClientId = data.senderId
+		clientId = data.receiverId
+		fromClientName = SearchMessageViewModel.getDisplayName(clientId: fromClientId, members: members)
+		message = data.message
+		dateCreated = data.createdTime
+		isMine = SearchMessageViewModel.isMine(senderID: fromClientId, ownerID: data.ownerClientId)
+		isQuoteMessage = SearchMessageViewModel.isQuote(message: data.message)
+		isForwardedMessage = SearchMessageViewModel.isForwarded(message: data.message)
+		isImageMessage = MessageUtils.isImageMessage(message: data.message)
+		isFileMessage = MessageUtils.isFileMessage(message: data.message)
+		clientWorkspaceDomain = data.ownerDomain
+		avatar = SearchMessageViewModel.getAvatar(clientId: fromClientId, members: members)
+		groupName = group
+	}
+
+	public func dateCreatedString() -> String {
+		return SearchMessageViewModel.dateString(from: dateCreated)
+	}
+
+	public func getQuoteMessage() -> String {
+		if isForwardedMessage {
+			return String(message.dropFirst(3))
+		}
+		let parts = message.dropFirst(3).split(separator: "|")
+		if parts.count == 4 {
+			return String(parts[1])
+		} else {
+			return message
+		}
+	}
+
+	public func getQuoteMessageReply() -> String {
+		let parts = message.dropFirst(3).split(separator: "|")
+		if parts.count == 4 {
+			return String(parts[3])
+		} else {
+			return ""
+		}
+	}
+
+	public func getQuoteMessageTime() -> String {
+		let parts = message.dropFirst(3).split(separator: "|")
+		if parts.count == 4 {
+			return String(parts[2])
+		} else {
+			return ""
+		}
+	}
+
+	public func getQuoteMessageName() -> String {
+		let parts = message.dropFirst(3).split(separator: "|")
+		if parts.count == 4 {
+			return String(parts[0])
+		} else {
+			return ""
+		}
+	}
+	
+	public func getQuoteMessageId() -> String {
+		return ""
+	}
+	
+	public func getQuoteDateString() -> String {
+		return ""
+	}
+}
+
+private extension SearchMessageViewModel {
+	static func dateString(from miliseconds: Int64) -> String {
+		let date = NSDate(timeIntervalSince1970: TimeInterval(miliseconds / 1000))
+		let formatDate = DateFormatter()
+		formatDate.dateFormat = "HH:mm"
+		return formatDate.string(from: date as Date)
+	}
+
+	static func getDisplayName(clientId: String, members: [IMemberModel]) -> String {
+		let member = members.first { member in
+			member.userId == clientId
+		}
+		return member?.userName ?? ""
+	}
+
+	static func getAvatar(clientId: String, members: [IMemberModel]) -> String {
+		let member = members.first { member in
+			member.userId == clientId
+		}
+		return member?.avatar ?? ""
+	}
+
+
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/ClearKeep/Scenes/Search/ViewModel/SearchMessageViewModel.swift#L124"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SearchMessageViewModel.swift:124</code></td>
    </tr>
  </tbody>
</table>